pkg/remote/trans/netpollmux/client_handler_test.go

```diff
@@ -108,3 +110,214 @@ func TestCliTransHandler(t *testing.T) {
+	ctx := context.Background()
+	ri := newMockRPCInfo()
+
+	buf := netpoll.NewLinkBuffer(1024)
+	npconn := &MockNetpollConn{
+		ReaderFunc: func() (r netpoll.Reader) {
+			return buf
+		},
+		WriterFunc: func() (r netpoll.Writer) {
+			return buf
+		},
+	}
+	conn := newMuxCliConn(npconn)
+
+	oneWayMsg := &MockMessage{
+		RPCInfoFunc: func() rpcinfo.RPCInfo {
+			return ri
+		},
+		ServiceInfoFunc: func() *serviceinfo.ServiceInfo {
+			return &serviceinfo.ServiceInfo{
+				Methods: map[string]serviceinfo.MethodInfo{
+					"method": serviceinfo.NewMethodInfo(nil, nil, nil, true),
+				},
+			}
+		},
+	}
+	err = handler.Write(ctx, conn, oneWayMsg)
+	test.Assert(t, err == nil, err)
+}
+
+// TestReadTimeout test client_handler read timeout because of no execute OnRequest
+func TestReadTimeout(t *testing.T) {
+	s := "hello world"
+	opt := &remote.ClientOption{
+		Codec: &MockCodec{
+			EncodeFunc: func(ctx context.Context, msg remote.Message, out remote.ByteBuffer) error {
+				r := mockHeader(msg.RPCInfo().Invocation().SeqID(), s)
+				n, err := out.WriteBinary(r.Bytes())
+				test.Assert(t, err == nil, err)
+				test.Assert(t, n == r.Len(), n, r.Len())
+				return err
+			},
+			DecodeFunc: func(ctx context.Context, msg remote.Message, in remote.ByteBuffer) error {
+				l := in.ReadableLen()
+				test.Assert(t, l == 3*codec.Size32+len(s))
+				in.Skip(3 * codec.Size32)
+				got, err := in.ReadString(len(s))
+				test.Assert(t, err == nil, err)
+				test.Assert(t, got == s, got, s)
+				return err
+			},
+		},
+		ConnPool: NewMuxConnPool(1),
+	}
+
+	handler, err := NewCliTransHandlerFactory().NewTransHandler(opt)
+	test.Assert(t, err == nil, err)
+
+	ctx := context.Background()
+	buf := netpoll.NewLinkBuffer(1024)
+
+	var isRead, isWrite bool
+
+	npconn := &MockNetpollConn{
+		ReaderFunc: func() (r netpoll.Reader) {
+			isRead = true
+			return buf
+		},
+		WriterFunc: func() (r netpoll.Writer) {
+			isWrite = true
+			return buf
+		},
+	}
+	conn := newMuxCliConn(npconn)
+
+	cfg := rpcinfo.NewRPCConfig()
+	rwTimeout := 1 * time.Second
+	rpcinfo.AsMutableRPCConfig(cfg).SetRPCTimeout(rwTimeout)
+
+	method := "method"
+	c := rpcinfo.NewEndpointInfo("", method, nil, nil)
+	to := rpcinfo.NewEndpointInfo("", method, nil, nil)
+	ri := rpcinfo.NewRPCInfo(c, to, rpcinfo.NewInvocation("", method), cfg, rpcinfo.NewRPCStats())
+	msg := &MockMessage{
+		RPCInfoFunc: func() rpcinfo.RPCInfo {
+			return ri
+		},
+		ServiceInfoFunc: func() *serviceinfo.ServiceInfo {
+			return &serviceinfo.ServiceInfo{
+				Methods: map[string]serviceinfo.MethodInfo{
+					"method": serviceinfo.NewMethodInfo(nil, nil, nil, false),
+				},
+			}
+		},
+	}
+
+	err = handler.Write(ctx, conn, msg)
+	test.Assert(t, err == nil, err)
+
+	time.Sleep(5 * time.Millisecond)
+	buf.Flush()
+	test.Assert(t, buf.Len() > len(s), buf.Len())
+	test.Assert(t, isWrite)
+
+	err = handler.Read(ctx, conn, msg)
+	test.Assert(t, rwTimeout <= trans.GetReadTimeout(ri.Config()))
+	test.Assert(t, err != nil, err)
```

Comment:
check the error type too

Reply:
the err used fmt.Errorf, not have error type or error code, I have modified it to match the err keyword